platform-hub-web/src/app/kubernetes/kubernetes-clusters-form.component.js

```diff
@@ -57,7 +57,7 @@ function KubernetesClustersFormController($state, KubernetesClusters, logger) {
     ctrl.saving = true;
 
     KubernetesClusters
-      .createOrUpdate(ctrl.cluster.id, ctrl.cluster)
+      .createOrUpdate(ctrl.cluster.name, ctrl.cluster)
```

Comment:
I guess this now needs to be split into separate `createX` vs `updateX` code paths (and corresponding methods in `KubernetesClusters` and `hubApiService`? (Check out the way it's done in the projects form component).